### PR TITLE
Avoid partial edit

### PR DESCRIPTION
**Pull request by jnwatson**
Prevent in-construction nodes from being retrieved from storage or the snap buid cache.
Cause fewer round trips to layer by consolidating changes relating to adding nodes or setting properties into "EditAtoms".

### CODE REVIEW

File: synapse/lib/base.py [hunk 1]
```diff
@@ -32,7 +32,8 @@ def _fini_atexit(): # pragma: no cover
             continue
 
         if not item._fini_atexit:
-            if __debug__:
+            # Nic tmp: don't check it
+            if 0 and __debug__:
```
Comment (jnwatson) [inline synapse/lib/base.py]: revert

File: synapse/tests/test_lib_snap.py [hunk 2]
```diff
@@ -75,6 +77,87 @@ class SnapTest(s_t_utils.SynTest):
...
+        ''' Test when a reader might retrieve a partially constructed node '''
+        NUM_TASKS = 2
+        failed = False
+        done_events = []
+        async with self.getTestCore() as core:
+
+            async def write_a_bunch(done_event):
+                nonlocal failed
+                data = list(range(50))
+                random.seed(4)  # chosen by fair dice roll
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_snap.py]: We should be using an a instance of `random.Random()` instead of the random module functions directly.

Comment (vEpiphyte) [reply synapse/tests/test_lib_snap.py]: the instantiation would have to be out of scope of the `write_a_bunch` function to preserve the current test behavior

Comment (vEpiphyte) [reply synapse/tests/test_lib_snap.py]: the current code shuffles the data into the same ordered list for each task.

File: synapse/lib/snap.py [hunk 3]
```diff
@@ -303,7 +306,10 @@ def setUser(self, user):
         try:
 
             fnib = self._getNodeFnib(name, valu)
-            return await self._addNodeFnib(fnib, props=props)
+            print(f'>{fnib[0].name}={fnib[1]}', flush=True)
```
Comment (invisig0th) [inline synapse/lib/snap.py]: should these be logger calls or removed?

Comment (jnwatson) [reply synapse/lib/snap.py]: they should be nuked.

Comment (vEpiphyte) [reply synapse/lib/snap.py]: :atomic:

**Pull request merged**